src/kani-compiler/src/main.rs

```diff
@@ -6,7 +6,7 @@
 //!
 //! Like miri, clippy, and other tools developed on the top of rustc, we rely on the
 //! rustc_private feature and a specific version of rustc.
-
+#![deny(warnings)]
```

Comment:
Hmm, normally I wouldn't want this, but since we control the exact Rust toolchain... lgtm?